diff --git a/xmtp_db/src/encrypted_store/schema_gen.rs b/xmtp_db/src/encrypted_store/schema_gen.rs
index c6fbf1c8..e10a8333 100644
--- a/xmtp_db/src/encrypted_store/schema_gen.rs
+++ b/xmtp_db/src/encrypted_store/schema_gen.rs
@@ -1,8 +1,10 @@
 // @generated automatically by Diesel CLI.

+use super::schema::conversation_list;
+
 diesel::table! {
     association_state (inbox_id, sequence_id) {
         inbox_id -> Text,
         sequence_id -> BigInt,
         state -> Binary,
     }
@@ -146,7 +148,8 @@ diesel::allow_tables_to_appear_in_same_query!(
     identity_updates,
     key_package_history,
     openmls_key_store,
     openmls_key_value,
     refresh_state,
     user_preferences,
+    conversation_list
 );
